SourceryFramework/Sources/Parsing/FileParser.swift

```diff
@@ -125,45 +125,46 @@ public final class FileParser {
         var typealiases = [Typealias]()
         try walkDeclarations(source: source) { kind, name, access, inheritedTypes, source, definedIn, next in
             let type: Type
-            switch kind {
-            case .protocol:
+            switch (kind, name) {
+            case (.protocol, .some(let name)):
```

Comment:
I suppose with latest versions of Swift `.some` or `?` is not needed to unwrap optional in switch, so could be just `let (.protocol, name)`?

Reply:
In this case `name` is optional:
```
let (.protocol, name)
```

Valid options are:

```
let (.protocol, name?)
(.protocol, let name?)
(.protocol, .some(let name))
(.protocol, let .some(name))
let (.protocol, .some(name))
```

What should I use? I can use first one if it suits codestyle (`let (.protocol, name?)`), it's the most contemporary syntax, I mean, it was added most recently.

Reply:
Good to know is whether it's working with old Xcode version. Every version I described works in Xcode 11.